src/main/java/com/google/sps/servlets/DataFromDatabaseServlet.java

```diff
@@ -42,7 +42,11 @@
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    //TODO: check to make sure there is userid or deviceid in request
+    if (request.getParameter("user_id") == null || request.getParameter("user_id") == "") {
```

Comment:
Consider extracting this into a helper method, isParameterEmpty(String parameterName), and inside it, you can check if the parameter is null or empty string. 

And here we can simply have  if(isParameterEmpty(user_id) && isParameterEmpty("device_id")).

Reply:
Done